internal/witness/witness.go

```diff
@@ -21,18 +21,39 @@ import (
 	"bytes"
 	"context"
 	"fmt"
+	"sync"
 
 	"github.com/transparency-dev/formats/log"
 	"github.com/transparency-dev/merkle"
 	"github.com/transparency-dev/merkle/compact"
 	"github.com/transparency-dev/merkle/proof"
 	"github.com/transparency-dev/witness/internal/persistence"
+	"github.com/transparency-dev/witness/monitoring"
 	"golang.org/x/mod/sumdb/note"
 	"google.golang.org/grpc/codes"
 	"google.golang.org/grpc/status"
 	"k8s.io/klog/v2"
 )
 
+var (
+	doOnce                         sync.Once
+	counterUpdateAttempt           monitoring.Counter
+	counterUpdateSuccess           monitoring.Counter
+	counterInvalidConsistency      monitoring.Counter
+	counterInconsistentCheckpoints monitoring.Counter
+)
+
+func initMetrics() {
+	doOnce.Do(func() {
+		mf := monitoring.GetMetricFactory()
+		const logIDLabel = "logid"
+		counterUpdateAttempt = mf.NewCounter("witness_update_request", "Number of attempted requests made to update checkpoints for the log ID", logIDLabel)
+		counterUpdateSuccess = mf.NewCounter("witness_update_success", "Number of successful requests made to update checkpoints for the log ID", logIDLabel)
+		counterInvalidConsistency = mf.NewCounter("witness_update_invalid_consistency", "Number of times the witness received a bad consistency proof for the log ID", logIDLabel)
+		counterInconsistentCheckpoints = mf.NewCounter("witness_update_inconsistency_checkpoints", "Number of times the witness received inconsistency checkpoints for the log ID", logIDLabel)
```

Comment:
```suggestion
		counterInconsistentCheckpoints = mf.NewCounter("witness_update_inconsistency_checkpoints", "Number of times the witness received inconsistent checkpoints for the log ID", logIDLabel)
```

Reply:
Fixed, and also in the more important place in the metric name :-)

Can I have another review now? Ta!